CRNN.py

```diff
@@ -193,35 +194,87 @@ def compute_output_shape(self, input_shape):
 
 
 if __name__ == "__main__":
+    input_shape = (28, 28, 1)
     model = tf.keras.models.Sequential()
+    model.add(tf.keras.layers.Conv2D(32, (3, 3), activation='relu', input_shape=input_shape))
+    model.add(tf.keras.layers.MaxPooling2D((2, 2)))
+
     model.add(RecurrentConvolutionalLayer(
         32,
         (3, 3),
         execution_depth=3,
-        input_shape=(28, 28, 1)
+        input_shape=input_shape
+    ))
+
+    model.add(RecurrentConvolutionalLayer(
+        32,
+        (3, 3),
+        execution_depth=3,
+        input_shape=input_shape
+    ))
+
+    model.add(tf.keras.layers.MaxPooling2D((2, 2)))
+
+    model.add(RecurrentConvolutionalLayer(
+        32,
+        (3, 3),
+        execution_depth=3,
+        input_shape=input_shape
+    ))
+    model.add(RecurrentConvolutionalLayer(
+        32,
+        (3, 3),
+        execution_depth=3,
+        input_shape=input_shape
     ))
-    model.summary()
+
+    model.add(tf.keras.layers.GlobalMaxPool2D())
 
     model.add(tf.keras.layers.Flatten())
-    model.add(tf.keras.layers.Dense(50, activation='relu'))
     model.add(tf.keras.layers.Dense(10, activation='softmax'))
+
     model.summary()
 
     model.compile(
         optimizer=tf.train.AdagradOptimizer(learning_rate=.1),
-        loss='sparse_categorical_crossentropy',
+        loss='kullback_leibler_divergence',
```

Comment:
why have we changed this? the orig. paper uses cross-entropy.

Reply:
Was a test. I will change it